### DISCUSSION TITLE
Notification and 1-click approval of timesheet outside of did

### DISCUSSION THREAD

**Discussion by pettersh** [Puzzlepart/did#912] [Ideas]
I propose a standard feature for notifying a did user about a timesheet to be confirmed outside of the did web ui. This could be easily done using an adaptive card in combination with the bot framework.

The notification card could be sent to the user in Outlook, in Teams and other places (i.e. Slack, Workplace, SharePoint) and the card contains a summary of the week with an [Confirm] button and a [Open in Did] button. The [Confirm] button allows the user to confirm without leaving the application where the notification/card is displayed.

**Comment by olemp** [comment]
It's a good idea, but I'm not sure how confirming would work outside Did. To confirm your hours, you need your events from Outlook. To get your events from Outlook you need to be authorized/logged in and get an access token (which has 1 hour expiry).

I'm not sure technically how this magic button would grab the events. match them, and register them/confirm them from outside Did context.

Notifications outside Did is not so far away though.. but just with one action - _open week in Did_.

Summoning @damsleth @okms

**Reply by okms** [reply]
I think a PoC made for Puzzlepart is step one. I absolutely see the niceness of this feature, but we do not know if this is something which should be part of the product. Im thinking this can start out as a sample Did integration first. But before that we need a PoC
